wallet/src/repository/wallet/mod.rs

```diff
@@ -495,6 +551,23 @@ where
                 _ => continue,
             };
 
+            // Check if tally txn corresponds to a wallet sent data request
+            if let types::Transaction::Tally(tally) = &txn {
+                // There is a DR transaction persisted in database whose tally was found or
+                // there is a DR transaction in pending state whose tally was found
+                if self
+                    .db
+                    .get::<_, u32>(&keys::transactions_index(tally.dr_pointer.as_ref()))
+                    .is_ok()
+                    || state
+                        .pending_dr_movements
+                        .contains_key(&tally.dr_pointer.to_string())
```

Comment:
Does it make sense to change the order of the condition here? So we first check if the transaction is in memory, and only check the database if it's not in memory.